app/products/calls/actions/permissions.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Alert, Platform} from 'react-native';
+import DeviceInfo from 'react-native-device-info';
+import Permissions from 'react-native-permissions';
+
+import type {IntlShape} from 'react-intl';
+
+const getMicrophonePermissionDeniedMessage = (intl: IntlShape) => {
+    const {formatMessage} = intl;
+    const applicationName = DeviceInfo.getApplicationName();
+    return {
+        title: formatMessage({
+            id: 'mobile.microphone_permission_denied_title',
+            defaultMessage: '{applicationName} would like to access your microphone',
+        }, {applicationName}),
+        text: formatMessage({
+            id: 'mobile.microphone_permission_denied_description',
+            defaultMessage: 'To participate in this call, open Settings to grant Mattermost access to your microphone.',
+        }),
+    };
+};
+
+export const hasMicrophonePermission = async (intl: IntlShape) => {
+    const targetSource = Platform.OS === 'ios' ? Permissions.PERMISSIONS.IOS.MICROPHONE : Permissions.PERMISSIONS.ANDROID.RECORD_AUDIO;
```

Comment:
Not sure what you mean--the ternary will always return a string.

Reply:
Addressed in 21a6712321dac2b79d8bace4a056adf10496e8f2
But for my own understanding, is there a reason to prefer the `Platform.select` over the ternary? Afaict, the ternary wouldn't return an `undefined` (which is what my `not sure what you mean` was about).

Reply:
It is slightly more intention revealing, and in my opinion easier to read. But it is mostly a personal preference.

The undefined problem is if you do something like `Platform.select({ios: 'foo', android:'foo'})`. In that case, since Platorm may be other, like web, that version of Platform.select would return type `string | undefined`. Setting the default makes it so only returns string.